source/mesh_deformation/fastscape.cc

```diff
@@ -1479,116 +1473,43 @@ namespace aspect
-            this->get_pcout() << "   Restarting runs with nonzero porosity can lead to a different system after restart. " << std::endl;
-          unsigned int line = 0;
-          while (line < fastscape_array_size)
-            {
-              in_silt_fraction >> silt_fraction[line];
-              line++;
-            }
-        }
 
-      // Now load the last output at time of restart.
-      // this allows us to correctly track when to call
-      // FastScape to make new VTK files.
-      std::ifstream in_last_output_time(restart_filename_time);
-      AssertThrow (in_last_output_time, ExcIO());
-      {
-        in_last_output_time >> last_output_time;
-      }
-    }
 
     template <int dim>
-    void FastScape<dim>::save_restart_files(const std::vector<double> &elevation,
-                                            std::vector<double> &basement,
-                                            std::vector<double> &silt_fraction) const
+    void
+    FastScape<dim>::load (const std::map<std::string, std::string> &status_strings)
     {
-      this->get_pcout() << "      Writing FastScape restart file... " << std::endl;
-
-      // Create variables for output directory and restart file
-      const unsigned int fastscape_array_size = fastscape_nx*fastscape_ny;
-      std::string dirname = this->get_output_directory();
-      const std::string restart_filename_elevation = dirname + "fastscape_elevation_restart.txt";
-      const std::string restart_filename_basement = dirname + "fastscape_basement_restart.txt";
-      const std::string restart_filename_silt_fraction = dirname + "fastscape_silt_fraction_restart.txt";
-      const std::string restart_filename_time = dirname + "fastscape_last_output_time.txt";
-
-      std::ofstream out_elevation(restart_filename_elevation);
-      std::ofstream out_basement(restart_filename_basement);
-      std::ofstream out_silt_fraction(restart_filename_silt_fraction);
-      std::ofstream out_last_output_time(restart_filename_time);
-      std::stringstream buffer_basement;
-      std::stringstream buffer_elevation;
-      std::stringstream buffer_silt_fraction;
-      std::stringstream buffer_time;
-
-      fastscape_copy_basement_(basement.data());
-
-      // If marine sediment transport and deposition is active,
-      // we also need to store the silt fraction.
-      if (use_marine_component)
-        fastscape_copy_f_(silt_fraction.data());
-
-      out_last_output_time << last_output_time << "\n";
-
-      for (unsigned int i = 0; i < fastscape_array_size; ++i)
+      if (status_strings.find("FastScape") != status_strings.end())
         {
-          buffer_elevation << elevation[i] << "\n";
-          buffer_basement << basement[i] << "\n";
-          if (use_marine_component)
-            buffer_silt_fraction << silt_fraction[i] << "\n";
+          std::istringstream is (status_strings.find("FastScape")->second);
+          aspect::iarchive ia (is);
+          ia >> (*this);
         }
```

Comment:
I suspect that we do this in a lot of places, and so this isn't a problem with your patch specifically: What happens if we *don't* find `FastScape` in `status_strings`? In that case, we can deserialize, and that's probably not good. Should we add 
```
else
   AssertThrow (false, "Trying to load data for FastScape from a checkpoint, but no data seems to have been written."))
```
or something like this?

Reply:
Added.